### ISSUE TITLE
move MLPClassifier to multilabel, multiclass

### ISSUE THREAD

**Issue by beckermr** [civisanalytics/muffnn#47]

**Comment by stephen-hoover** [comment]
I've discovered that the MLPClassifier does not encode string targets when y is 2D. Is that part of this issue, or should I file a new issue? It's not clear what this issue covers.

**Comment by mheilman** [comment]
That sounds like a separate issue.

@beckermr, could you clarify what this issue means?  I believe it means adding support for multilabel-multiclass modeling.  That right?

**Comment by beckermr** [comment]
This was me thinking of using the list of lists format from sklearn. @stephen-hoover's question is a different issue.

**Comment by beckermr** [comment]
@stephen-hoover Is that format supposed to be supported by sklearn API? I am reading the docs and it is very unclear.

**Comment by stephen-hoover** [comment]
The reason to handle encoding for multilabel inputs is so that we can do multilabel-multiclass, so this issue sounds like it's still a blocker. 

@beckermr , I agree it's unclear. Experimentally, the tree ensemble classifiers do correctly handle string input for multilabel-multiclass problems. There's a bug in the input checking which causes an error for target arrays with mixed string and numeric columns, and the `predict` method errors if you used string labels in the training. However `fit` and `predict_proba` work. I think we should support it in `muffnn`; it seems like a common use case to me.
